kola/tests/docker/docker.go

```diff
@@ -459,39 +492,98 @@ func dockerUserNoCaps(c cluster.TestCluster) {
 	}
 }
 
-// testDockerInfo test that docker info's output is as expected.  the expected
-// filesystem may be asserted as one of 'overlay', 'btrfs', 'devicemapper'
-// depending on how the machine was launched.
-func testDockerInfo(expectedFs string, c cluster.TestCluster) {
+// dockerContainerdRestart ensures containerd will restart if it dies. It tests that containerd is running,
+// kills it, the tests that it came back up.
+func dockerContainerdRestart(c cluster.TestCluster) {
 	m := c.Machines()[0]
 
-	dockerInfoJson, err := m.SSH(`curl -s --unix-socket /var/run/docker.sock http://docker/v1.24/info`)
+	pid, err := m.SSH("systemctl show containerd -p MainPID --value")
 	if err != nil {
-		c.Fatalf("could not get dockerinfo: %v", err)
+		c.Fatal(err)
+	}
+	if string(pid) == "0" {
+		c.Fatalf("Could not find containerd pid")
 	}
 
-	type simplifiedDockerInfo struct {
-		ServerVersion string
-		Driver        string
-		CgroupDriver  string
-		Runtimes      map[string]struct {
-			Path string `json:"path"`
-		}
-		ContainerdCommit struct {
-			ID       string
-			Expected string
+	// verify containerd came up
+	if !testContainerdUp(c) {
+		c.Fatalf("Containerd did not start")
+	}
+
+	// kill it
+	if _, err = m.SSH("sudo kill " + string(pid)); err != nil {
+		c.Fatal(err)
+	}
+
+	// retry polling its state
+	util.Retry(12, 6*time.Second, func() error {
+		state, err := m.SSH("systemctl show containerd -p SubState --value")
+		if err != nil {
+			c.Fatal(err)
 		}
-		RuncCommit struct {
-			ID       string
-			Expected string
+		switch string(state) {
+		case "running":
+			return nil
+		case "stopped", "exited", "failed":
+			c.Fatalf("containerd entered stopped state")
 		}
-		SecurityOptions []string
+		return fmt.Errorf("containerd failed to restart")
+	})
+
+	// verify systemd started it and that it's pid is different
+	newPid, err := m.SSH("systemctl show containerd -p MainPID --value")
+	if err != nil {
+		c.Fatal(err)
+	}
+	if string(newPid) == "0" {
+		c.Fatalf("Containerd is not running (could not find pid)")
+	} else if string(newPid) == string(pid) {
+		c.Fatalf("Old and new pid's are the same. containerd did not die")
+	}
+
+	// verify it came back and docker knows about it
+	if !testContainerdUp(c) {
+		c.Fatalf("Docker could not find containerd")
+	}
+
+}
+
+func testContainerdUp(c cluster.TestCluster) bool {
```

Comment:
More bikeshedding; this doesn't have to return a bool now that there's no attempt to observe containerd not being up.

It can just `c.Fatalf` itself if the commits don't match. Arguably `assert` is clearer than `test` when the helper is doing that, but I know I already started the pattern of using `test` in this file.